### PR TITLE
Warn users when installing plugins outside of Twilio and Twilio-Labs

### PR DESCRIPTION
**Pull request by thinkingserious**
### Checklist
* [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: src/hooks/init/plugin-install.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    logger.warn('WARNING!!! You are attempting to install a plugin from an untrusted source.');
+    logger.warn('It could contain malicious software or in other ways compromise your system.');
+    let response = await prompt([{
+      type: 'confirm',
+      name: 'continue',
+      message: 'Are you sure you want to continue?',
+      default: false
+    }]);
+    if (response.continue === false) {
+      this.exit();
```
Comment (childish-sambino) [inline src/hooks/init/plugin-install.js]: Should be non-zero exit code. 1 is fine.

File: src/hooks/init/plugin-install.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+const prompt = require('inquirer').prompt;
```
Comment (childish-sambino) [inline src/hooks/init/plugin-install.js]: Need to add inquirer as direct dependency.

File: src/hooks/init/plugin-install.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+const { logger } = require('@twilio/cli-core').services.logging;
+const ALLOWED_ORGS = [
+  '@twilio',
+  '@twilio-labs'
```
Comment (childish-sambino) [inline src/hooks/init/plugin-install.js]: Should use trailing `/` so someone doesn't just make an `@twilio-schmilio` org that passes.

File: src/hooks/init/plugin-install.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+];
+
+function isTwilioPlugin(options) {
+  let isValid = false;
+  if (options.plugin.name === undefined) {
+    return isValid;
+  }
+  ALLOWED_ORGS.forEach(function (org) {
```
Comment (childish-sambino) [inline src/hooks/init/plugin-install.js]: Could use `.find` to short-circuit execution.

File: src/hooks/init/plugin-install.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    if (options.plugin.name.startsWith(org)) {
+      isValid = true;
+    }
+  });
+  return isValid;
+}
+
+module.exports = async function (options) {
```
Comment (childish-sambino) [inline src/hooks/init/plugin-install.js]: Need tests.

File: src/hooks/init/plugin-install.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+    const prompt = require('inquirer').prompt;
+    const response = await prompt([{
+      type: 'confirm',
+      name: 'continue',
+      message: 'Are you sure you want to continue?',
+      default: false
+    }]);
+    if (response.continue === false) {
+      process.exit(1);
```
Comment (childish-sambino) [inline src/hooks/init/plugin-install.js]: I think `this.exit(1)` works, just need to mock it in the tests like `ctx.exit = sinon.stub();`.

Comment (thinkingserious) [reply src/hooks/init/plugin-install.js]: `this.exit(1)` throws an error.

Comment (thinkingserious) [reply src/hooks/init/plugin-install.js]: I get `TypeError: this.exit is not a function`

Comment (thinkingserious) [reply src/hooks/init/plugin-install.js]: OK, mocking out ctx.exit appears to work. Just pushed the changes. 👍

File: src/hooks/init/plugin-install.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+/* eslint-disable no-process-exit */
+/* eslint-disable unicorn/no-process-exit */
```
Comment (childish-sambino) [inline src/hooks/init/plugin-install.js]: These are no longer needed.

**Pull request merged**